pandas/tools/pivot.py

```diff
@@ -434,12 +437,41 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
     data.update(zip(rownames, index))
     data.update(zip(colnames, columns))
 
+    if values is None and aggfunc is not None:
+        raise ValueError("`aggfunc` cannot be used with `values`.")
+
+    if normalize is not None and (aggfunc is not None or values is not None):
+        error = "`normalize` cannot be combined with `values` or `aggfunc`"
+        raise ValueError(error)
+
     if values is None:
         df = DataFrame(data)
         df['__dummy__'] = 0
         table = df.pivot_table('__dummy__', index=rownames, columns=colnames,
                                aggfunc=len, margins=margins, dropna=dropna)
-        return table.fillna(0).astype(np.int64)
+
+        if normalize is not None:
+            if normalize == 'all':
+                table = table / table.sum(axis=1).sum(axis=0)
+
+            elif normalize == 'rows':
+                table = table.apply(lambda x: x / x.sum(), axis=1)
```

Comment:
in any event this is just: ``table/x.sum(1)``. we almost *never* actually want to use ``.apply``